src/movement/CommunityBasedMovement.java

```diff
@@ -0,0 +1,227 @@
+/*
+ * Copyright 2010 Aalto University, ComNet
+ * Released under GPLv3. See LICENSE.txt for details.
+ */
+
+/**
+ * CommunityBasedMovement where the coordinates are restricted to rectangle
+ * area defined by two coordinates in the entire map.
+ * 
+ * Four parameters for configuration file:
+ * CommunityBasedMovement.communityNumber: the number of rectangle community
+ *		e.g. if you have 12 communities, you could set the value to 3,4 or 4,3
+ * CommunityBasedMovement.mapSize: the size of map should be identical to worldSize
+ * 		e.g. CommunityBasedMovement.mapSize = x,y : x,y are the maximum coordinates
+ * 			in X-Axis and Y-Axis respectively 
+ * CommunityBasedMovement.communityID = a,b,c,d: all community identity
+ * CommunityBasedMovement.probabilities_local_roaming: probabilities for p_local and 
+ * 		p_roaming
+ * 
+ * Note: To run this movement model:
+ * 1. Move the configuration file "communitybasedmovement_settings.txt" from 
+ * the directory "example_settings" to root directory
+ * 2.rename the configuration to "default_settings.txt" to replace the original
+ * one
+ * 3. recompile the code
```

Comment:
Dear akeranen, 
What i wrote is the source file .java not just a configuration file. So, you have to recompile the new movement model source file into the project. But there is not any conflicts against previous source code and no any modification in previous source code. I just added the new movement model file and explained how to configure the configuration file to test it.